core/tests/protractor_utils/editor.js

```diff
@@ -227,6 +228,152 @@ var expectCannotDeleteInteraction = function() {
     '.protractor-test-delete-interaction')).isPresent()).toBeFalsy();
 };
 
+// GADGETS
+
+// Additional arguments may be sent to this function, and they will be
+// passed on to the relevant gadget editor.
+var addGadget = function(panelName, gadgetType, gadgetName) {
+
+  // Bring up the gadget insertion modal for the specified panel.
+  element(
+    by.css('.protractor-test-' + panelName + '-panel-insert-gadget-button'))
+    .click();
+
+  general.waitForSystem(2000);
+
+  // Select the desired gadgetType from the modal.
+  element(
+    by.css('.protractor-test-' + gadgetType + '-gadget-selection-modal'))
+    .click();
+
+  var gadgetNameInput = element(by.css('.protractor-test-gadget-name-input'));
+  gadgetNameInput.clear().then(function() {
+    gadgetNameInput.sendKeys(gadgetName);
+  });
+
+  // Locate the customization section and apply any customizations.
+  var elem = element(by.css('.protractor-test-gadget-customization-editor'));
+  var customizationArgs = [elem];
+
+  if (arguments.length > 3) {
+    for (var i = 3; i < arguments.length; i++) {
+      customizationArgs.push(arguments[i]);
+    }
+    gadgets.getGadget(gadgetType).customizeGadget.apply(
+      null, customizationArgs);
+  }
+
+  element(by.css('.protractor-test-save-gadget-button')).click();
+
+  // Wait for the customization modal to close.
+  general.waitForSystem(2000);
+
+};
+
+// currentName must exist.
```

Comment:
Please be more explicit: what does 'exist' mean? I'd expect something along the lines of:

Callers should ensure that [...something involving currentName...]